core/scripts/cli/sponsor/currencyUtils.js

```diff
@@ -14,7 +14,23 @@ const unwrapToEth = async (web3, artifacts, emp, amount) => {
   await submitTransaction(web3, async () => await weth.withdraw(amount.toString()), "Unwrapping WETH to ETH");
 };
 
+const getIsWeth = async (web3, artifacts, collateralCurrency) => {
+  const WETH9 = artifacts.require("WETH9");
+  return collateralCurrency.address === WETH9.address;
+};
+
+const getCurrencySymbol = async (web3, artifacts, collateralCurrency) => {
+  if (await getIsWeth(web3, artifacts, collateralCurrency)) {
+    return "WETH";
+  } else {
+    // TODO: Do all collateral currencies we care about support `symbol()`?
```

Comment:
I wouldn't assume that, but maybe you could check if `symbol()` returns something?

Reply:
The stupider option I was considering was just hardcoding specific addresses in prod. E.g., we know the DAI address, etc. We can revisit this later too.

Reply:
I think this is a fair assumption that they will all have it. DAI and USDC certially do any I am yet to see an ERC20 token that does not have it.

Reply:
We could also do something simple like wrap `symbol()` in a try catch, and just show `token 0x1234...5678` if it fails. Later if we want good support for a token that doesn't support symbol, we just add some hardcoded addresses that get checked first.

Reply:
I'll do that in a follow up.